Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XLGToDBAnalyzer.cs

```diff
@@ -389,24 +470,155 @@ public override void DominoInvocation(DominoInvocationEventData data)
         public override void PipExecutionDirectoryOutputs(PipExecutionDirectoryOutputs data)
         {
             var pipExecDirectoryOutputEvent = data.ToPipExecutionDirectoryOutputsEvent(WorkerID.Value, PathTable);
-            var eq = new Xldb.EventTypeQuery
+            var eq = new EventTypeQuery
             {
                 EventTypeID = Xldb.ExecutionEventId.PipExecutionDirectoryOutputs,
-                UUID = pipExecDirectoryOutputEvent.UUID
+                UUID = Guid.NewGuid().ToString()
             };
 
-            WriteToDb(eq.ToByteArray(), pipExecDirectoryOutputEvent.ToByteArray());
+            WriteToDb(eq.ToByteArray(), pipExecDirectoryOutputEvent.ToByteArray(), XldbDataStore.EventColumnFamilyName);
+        }
+
+        /// <summary>
+        /// Ingest all of the pips to RocksDB
+        /// </summary>
+        public void IngestAllPips()
+        {
+            var totalNumberOfPips = CachedGraph.PipTable.StableKeys.Count;
+            var pipIds = CachedGraph.PipTable.StableKeys;
+            var concurrency = Environment.ProcessorCount;
+            var partitionSize = totalNumberOfPips / concurrency;
+            Console.WriteLine("Ingesting pips now ...");
+
+            Parallel.For(0, concurrency, i =>
+            {
+                var start = i * partitionSize;
+                var end = (i + 1) == concurrency ? totalNumberOfPips : (i + 1) * partitionSize;
+                var pipsIngested = 0;
+                var pipSemistableKeys = new List<byte[]>();
+                var pipSemistableValues = new List<byte[]>();
+                var pipPipIdKeys = new List<byte[]>();
+                var pipValues = new List<byte[]>();
+
+                for (int j = start; j < end; j++)
+                {
+                    var pipId = pipIds[j];
+                    pipsIngested++;
+
+                    if (pipsIngested % 100000 == 0)
+                    {
+                        Console.Write(".");
+                        WriteBatchToDb(pipSemistableKeys, pipSemistableValues, XldbDataStore.PipColumnFamilyName);

```

Comment:
You are doing 2 separate transactions here, may want to do a single one, or do the 2 in parallel. Also, if you are anyhow able to feed the keys in sorted order, that should speed up insertion.